readthedocs/core/signals.py

```diff
@@ -0,0 +1,16 @@
+from corsheaders import signals
+from readthedocs.projects.models import Domain
+
+from urlparse import urlparse
+
+
+def handler(sender, request, **kwargs):
+    whitelist_urls = ['/api/v2/footer_html', '/api/v2/search']
+    host = urlparse(request.META['HTTP_ORIGIN']).netloc.split(':')[0]
+    domains = Domain.objects.filter(domain__icontains=host)
+    for url in whitelist_urls:
+        if (request.path_info.startswith(url) and domains.count()):
```

Comment:
Perhaps `domains.count()` can be moved outside of the loop? If `domains.count()` returns 0, we can just return `False`.